spl-balance/src/lib.rs

```diff
@@ -0,0 +1,178 @@
+                            }) => {
+                                log::info!("Instruction: InitializeMint");
+                                output.set(
+                                    0,
+                                    format!(
+                                        "address:{}",
+                                        bs58::encode(msg.account_keys[inst.accounts[0] as usize])
+                                            .into_string()
+                                    ),
+                                    &get_token(
+                                        msg.account_keys[inst.accounts[0] as usize].to_vec(),
+                                        decimals,
+                                        mint_authority,
+                                        freeze_authority,
+                                        bs58::encode(&transaction.signatures[0]).into_string(),
+                                        block.block_height.as_ref().unwrap().block_height,
+                                    ),
+                                );
+                            }
+                            Ok(instruction::TokenInstruction::InitializeMint2 {
+                                decimals,
+                                mint_authority,
+                                freeze_authority,
+                            }) => {
+                                log::info!("Instruction: InitializeMint2");
+                                output.set(
+                                    0,
+                                    format!(
+                                        "address:{}",
+                                        bs58::encode(msg.account_keys[inst.accounts[0] as usize])
+                                            .into_string()
+                                    ),
+                                    &get_token(
+                                        msg.account_keys[inst.accounts[0] as usize].to_vec(),
+                                        decimals,
+                                        mint_authority,
+                                        freeze_authority,
+                                        bs58::encode(&transaction.signatures[0]).into_string(),
+                                        block.block_height.as_ref().unwrap().block_height,
+                                    ),
+                                );
+                            }
+                            _ => {}
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+// helper to create a TokenAccount entity
+fn get_token(
+    token_address: Vec<u8>,
+    decimal: u8,
+    mint_authority: Vec<u8>,
+    freeze_authority_opt: Option<Vec<u8>>,
+    tx_id: String,
+    block_height: u64,
+) -> proto::TokenAccount {
+    let mut token = proto::TokenAccount {
+        address: bs58::encode(&token_address).into_string(),
+        name: "".to_string(),
```

Comment:
Think all protos inherit the default trait automatically so you might be able to get away with ..default at the bottom rather than manually spefying these fields during the instantiation